frontend/packages/dev-console/src/components/edit-application/edit-application-utils.ts

```diff
@@ -175,6 +175,31 @@ export const getKsvcRouteData = (resource: K8sResourceKind) => {
   return routeData;
 };
 
+export const getDefaultLabels = () => {
+  return [
+    'app',
+    'app.kubernetes.io/instance',
+    'app.openshift.io/runtime',
+    'app.kubernetes.io/part-of',
+    'app.openshift.io/runtime-version',
+    'app.openshift.io/runtime-namespace',
+    'networking.knative.dev/visibility',
+  ];
+};
+
+export const getRouteLabels = (
+  route: K8sResourceKind,
+  resource: K8sResourceKind,
+): Record<string, string> => {
+  const allLabels = _.get(resource, 'metadata.labels', {});
+  const allRouteLabels = _.get(route, 'metadata.labels', {});
+  const filteredRouteLabels = _.omit(allRouteLabels, [
+    ...getDefaultLabels(),
+    ...Object.keys(allLabels),
```

Comment:
@jerolimov `allLabels` will have the `defaultLabels` as well so there is no need for adding `defaultLabels` twice.

Reply:
@divyanshiGupta Technical you are right here, but I think it's good to remove everything from our sourcecode driven list as well. For example, if we change or add another label here. If you not 100% against this I would like to keep it. Wdyt? :)